src/mscorlib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -381,26 +381,34 @@ private static bool TryLoadTzFile(string tzFilePath, ref byte[] rawData, ref str
         /// </summary>
         private static string FindTimeZoneIdUsingReadLink(string tzFilePath)
         {
-            string id = null;
-
-            StringBuilder symlinkPathBuilder = StringBuilderCache.Acquire(Path.MaxPath);
-            bool result = Interop.GlobalizationInterop.ReadLink(tzFilePath, symlinkPathBuilder, (uint)symlinkPathBuilder.Capacity);
-            if (result)
+            int capacity = 4096;
+            do
             {
-                string symlinkPath = StringBuilderCache.GetStringAndRelease(symlinkPathBuilder);
-                // time zone Ids have to point under the time zone directory
-                string timeZoneDirectory = GetTimeZoneDirectory();
-                if (symlinkPath.StartsWith(timeZoneDirectory))
+                StringBuilder symlinkPathBuilder = StringBuilderCache.Acquire(capacity);
+                int result = Interop.GlobalizationInterop.ReadLink(tzFilePath, symlinkPathBuilder, (uint)symlinkPathBuilder.Capacity);
+                if (result > 0 && result < symlinkPathBuilder.Capacity)
                 {
```

Comment:
you can check result < 0 once here and don't have to check result > 0 two times later